### PR TITLE
Issue #1824: Replaced most remaining sprintf with safer snprint

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Review by byrnHDF — APPROVED** [review]

**Comment by seanm** [comment]
@byrnHDF thanks for your speedy review!

Be sure the check the non-trivial ones carefully, they can be tricky...

File: test/filter_plugin.c
Comment (jhendersonHDF) [inline test/filter_plugin.c:1]: For the changes here, see #4001. Seems the two calls can simply be combined into a single `snprintf` call

Comment (seanm) [reply test/filter_plugin.c:1]: Oh funny timing that @hyoklee was also addressing some of these.  I'll integrate that here.

File: test/filter_plugin.c [hunk 1]
```diff
@@ -847,10 +847,7 @@ test_creating_groups_using_plugins(hid_t fid)
 
     /* Create multiple groups under the top-level group */
     for (i = 0; i < N_SUBGROUPS; i++) {
-        char *sp = subgroup_name;
```
Comment (jhendersonHDF) [inline test/filter_plugin.c:850]: Will need to remove the references to `sp` as well since that var. just goes away

Comment (seanm) [reply test/filter_plugin.c:850]: oh whoops I had that staged but for got to commit it...

**Comment by seanm** [comment]
For the last remaining one, I have a starting point: https://github.com/HDFGroup/hdf5/pull/4007

File: src/H5FDsplitter.c [hunk 2]
```diff
@@ -532,27 +532,28 @@ H5FD__splitter_get_default_wo_path(char *new_path, size_t new_path_len, const ch
...
     /* Determine if filename contains a ".h5" extension. */
-    if ((file_extension = strstr(base_filename, ".h5"))) {
+    file_extension = strstr(base_filename, ".h5");
+    if (file_extension) {
         /* Insert the suffix between the filename and ".h5" extension. */
-        strcpy(new_path, base_filename);
-        file_extension = strstr(new_path, ".h5");
-        sprintf(file_extension, "%s%s", suffix, ".h5");
+        intptr_t beginningLength = file_extension - base_filename;
+        snprintf(new_path, new_path_len, "%.*s%s%s", (int)beginningLength, base_filename, suffix, ".h5");
```
Comment (jhendersonHDF) [inline src/H5FDsplitter.c:539]: Ah clever and always nicer to do things in a single `snprintf` call!

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**